base_local_planner/include/base_local_planner/obstacle_cost_function.h

```diff
@@ -61,13 +61,13 @@ class ObstacleCostFunction : public TrajectoryCostFunction {
 
   void setSumScores(bool score_sums){ sum_scores_=score_sums; }
 
-  void setParams(double max_trans_vel, double max_forward_inflation, double max_sideward_inflation, double scaling_speed);
+  void setParams(double max_trans_vel, double max_forward_inflation, double max_sideward_inflation, double scaling_speed, bool occdist_use_footprint);
```

Comment:
How about using this instead? since we don't want to change the default behavior in base_local_planner right?

```suggestion
  void setParams(double max_trans_vel, double max_forward_inflation, double max_sideward_inflation, double scaling_speed, bool occdist_use_footprint = true);
```

Reply:
I guess we could do that, but this critic is only used for DWA so I don't think there's much point.

Reply:
The same argument as below, we shouldn't change the default behavior here because maybe we might want to update ros-navigation pkg in Sootballs in the future for bug fixes, etc. and the changes should be governed by params where ever we can.

Reply:
But we're not changing default behavior here -- in fact, we're explicitly asking the user which occdist mode they want to use. I think that's better than using default arguments and hiding the functionality. And since this is only used by DWA we're not breaking any code either.

But I agree with the other discussion that we should set the dynamic param default to current behavior.